parsec/core/fs/storage/workspace_storage.py

```diff
@@ -152,6 +156,15 @@ async def clear_chunk(self, chunk_id: ChunkID, miss_ok: bool = False) -> None:
             if not miss_ok:
                 raise
 
+    # Pattern filter interface
+
+    def get_pattern_filter(self) -> Pattern:
+        assert self._pattern_filter is not None
```

Comment:
`self._pattern_filter` is defined as Optional and is initialized to `None`...

I guess the thing is we have `_load_pattern_filter` that is called after `__init__` which set `_pattern_filter`.
In such case I think we should only declare the type in `__init__`:
```python
def __init__(self, [...]):
    [...]
    self._pattern_filter: Pattern  # Defined by BaseWorkspaceStorage._pattern_filter as part of init

async def run(self, [...]):
    self._pattern_filter = turbo_smurfing_the_stuff()
```

@AntoineDupre you did this kind of trick something in a previous PR but I couldn't remember where...